packages/server-wallet/src/handlers/join-channel.ts

```diff
@@ -29,11 +34,13 @@ export class JoinChannelError extends Error {
 const hasStateSignedByMe = (cs: ChannelState): boolean => !!cs.latestSignedByMe;
 
 const ensureNotSignedByMe = (cs: ChannelState): StepResult =>
-  hasStateSignedByMe(cs) ? left(new JoinChannelError(Errors.alreadySignedByMe)) : right(cs);
+  hasStateSignedByMe(cs)
+    ? left(new JoinChannelError(JoinChannelError.reasons.alreadySignedByMe))
+    : right(cs);
 
 function ensureLatestStateIsPrefundSetup(cs: ChannelState): StepResult {
   if (cs.latest.turnNum === 0) return right(cs);
-  return left(new JoinChannelError(Errors.invalidTurnNum));
+  return left(new JoinChannelError(JoinChannelError.reasons.invalidTurnNum));
```

Comment:
The only thing that would make this better is if you could do
```suggestion
  return left(JoinChannelError.invalidTurnNum());
```

Reply:
Agreed. We can definitely create these factories at least for often instantiated errors.